src/app/features/public-form/public-form.component.ts

```diff
@@ -86,20 +86,25 @@ export class PublicFormComponent<E extends Entity> implements OnInit {
   }
 
   private async loadFormConfig() {
-    const id = this.route.snapshot.paramMap.get("id");
+    try {
```

Comment:
While a general try-catch around everything works, of course, this is not very fine-grained error handling :wink:  

 Can you refine this to only "try" the part that can actually throw errors and also distinguish between "not found" (--> just display feedback to user) and any truly unexpected error (--> use Logging to track this in our monitoring)?

Reply:
i have updated the error handling to use Logging to track the unexpected error and for expected (Public form config not found ) displaying the feedback to user

Reply:
The way this is implemented now, I do not see any "expected" way why this should throw an error. Even without permissions, the loadAll would give an empty array, I think. So we can skip the try block completely - or am I missing something?

Generally, try-blocks should only wrap 1-2 lines of code, in my opinion. It should be very clear what kind of error can happen from exactly which line and then catch that specifically only.

Reply:
Previously, on my local setup(without docker usage), when the database was not connected, only the title and description were loading, and the fields were not visible. That’s why I thought adding a try-catch block to handle such cases.

However, I agree that we can skip this since no errors have occurred after establishing the database connection(using aam-services).

Also thanks for sharing the insights on try-catch usage, I'll keep this in mind going forward.

Reply:
those errors (preventing the fields to load) were caused by server not being reachable? Or what were the specific errors?
There could be some cases that we do want to catch. If so, let's add a try-catch but trying to be rather specific what we think could fail.

Reply:
Oh, right! That's a good point that we could maybe check and add an explicit error message for. I bet that'll save us a lot of time troubleshooting, the next time we run into this :laughing: 

How about injecting `EntityAbility` and using `this.ability.cannot("create", entityType)` to explicitly check the required permissions in our init method here, possibly setting an error like "You do not have the required permissions to submit this form. Please log in or contact your system administrator."?

Reply:
I initially injected `EntityAbility and AlertService` to check permissions and set errors within the init method. However, this approach always displayed an error, regardless of whether the permissions were set correctly or not. To resolve this, I moved the logic `this.ability.cannot("create", entityType)` into the loadFromConfig method, where it now works as expected.